src/main/java/de/retest/recheck/ui/descriptors/AttributeUtil.java

```diff
@@ -52,6 +54,10 @@ public static Rectangle getAbsoluteOutline( final IdentifyingAttributes attribut
 	}
 
 	private static Rectangle getActualOutline( final ElementDifference difference, final String type ) {
```

Comment:
The method is called from two public methods, please add tests for the fixed behavior.

Reply:
I think a method named `getIdentifyingAttributesDifference` not returning `IdentifyingAttributesDifference` is not intended behavior. Therefore I would not add a test to cement this further.

Reply:
When you say this can / did cause bugs, there shall be tests. There is really no need to discuss this further.

Reply:
Well. Adding a sensible test isn't easy at it involves downstream recheck projects. Adding just a mock test is cementing unwanted behavior.

Reply:
So what do you suggest?

Reply:
https://retest.atlassian.net/browse/RET-2015

Reply:
It is recheck, why is there a need to involve anything else?

You can have two simple unit tests: Invoke both `getActualOutline` and `getActualAbsoluteOutline` with an `ElementDifference` where `getIdentifyingAttributesDifference()` does _not_ return an `IdentifyingAttributesDifference` but something else, then you get the `ClassCastException` you just fixed?

We have a lot of "unwanted behavior" like this. That doesn't mean there shouldn't be tests for it.

Reply:
> I think a method named getIdentifyingAttributesDifference not returning IdentifyingAttributesDifference is not intended behavior. Therefore I would not add a test to cement this further.

But isn't this fix then only a workaround? Wouldn't the actual fix fix the cause on why the method does not return the said class (because it either is `null` or actually a wrong instance)?

On the other hand, if this is wanted behavior and the method is just poorly named, I agree with @beatngu13 to add a test for that. At least to make sure that this is not "broken" while we work on the actual fix, which then can remove/modify the test added here.

Reply:
An actual fix is probably way more complicated. I just added a test for that...